### PR TITLE
Example PR: Auto provision an EC2 Linux instance

### PR DESCRIPTION
**Pull request by zaro0508**
This is an example PR (do not merge!). It is a template to show
how to get the CI system to automatically provision an EC2 instance.

Instance creation workflow:

* Scientist create a PR similar to this one. Create a new file
* with a unique stack_name and updated parameters.
* Have someone review and approve the PR.
* Merge the PR
* The CI system will auto provision an EC2 instance
* An email will be sent to the owner email address with the new bucket info.

The newly created instance will contain the following tags:
```
key: Department
value: Platform

key: OwnerEmail
value: joe.smith@sagebase.org

key: Project
value: Infrastructure

key: aws:cloudformation:logical-id
value: Ec2Instance

key: aws:cloudformation:stack-id
value: arn:aws:cloudformation:us-east-1::stack/kickstart-ec2-service/559e55a0-bc47-11e8-995a-500c28b4b099

key: aws:cloudformation:stack-name
value: iwanna-instance
```

To remove the instance:

* Instance owner must create a new PR that deletes the files that was added by
the PR that was used to add the instance.
* Have an an AWS account Admin review your change.
* Approve and merge.

### CODE REVIEW

**Comment by thomasyu888** [comment]
Use this AMI id: ami-0ddee041772c2d9f8 to spin up an ec2 with an encrypted boot volume.

**Comment by thomasyu888** [comment]
Add VolumeSize to example

File: config/prod/iwanna-instance.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  # The Sage deparment for this resource
+  Department: "Platform"
+  # The Sage project this resource will be used for
+  Project: "Infrastructure"
+  # The resource owner
+  OwnerEmail: "joe.smith@sagebase.org"
+  # (Optional) EC2 instance type, default is "t2.nano" (other available types https://aws.amazon.com/ec2/instance-types/)
+  InstanceType: "t2.medium"
+  # (Optional) EC2 based AMI, default is "ami-0ddee041772c2d9f8" (AWS linux with 8 GB encrypted volume)
+  # Sage IT managed AMIs https://github.com/Sage-Bionetworks/scicomp-provisioner#ec2-amis
```
Comment (thomasyu888) [inline config/prod/iwanna-instance.yaml]: Remove these.

**Comment by thomasyu888** [comment]
I think it would be beneficial for us to define the department names, like
`Platform`, `CompOnc`, `SysBio`, `Governance`...

File: config/prod/iwanna-instance.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+parameters:
+  # The Sage deparment for this resource
+  Department: "Platform"
+  # The Sage project this resource will be used for
+  Project: "Infrastructure"
+  # The resource owner
+  OwnerEmail: "joe.smith@sagebase.org"
+  # (Optional) EC2 instance type, default is "t2.nano" (other available types https://aws.amazon.com/ec2/instance-types/)
+  InstanceType: "t2.medium"
+  # (Optional) EC2 boot volume size in GB, default is 8GB, Max is 1000GB
```
Comment (thomasyu888) [inline config/prod/iwanna-instance.yaml]: Should say here that it was to be a string

**Comment by thomasyu888** [comment]
Should still document the ability to specify `AMIId:`

File: config/prod/iwanna-instance.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  Department: "Platform"
+  # The Sage project (Infrastructure, amp-ad, ntap, dream, etc..)
+  Project: "Infrastructure"
+  # The resource owner
+  OwnerEmail: "joe.smith@sagebase.org"
+
+  # Default account settings, leave alone unless you plan to override
+  VpcName: "computevpc"
+  KeyName: "scicomp"
+  AMIId: "ami-0de53d8956e8dcf80"      # Amazon Linux 2 AMI 2.0.20190313
```
Comment (thomasyu888) [inline config/prod/iwanna-instance.yaml]: @zaro0508, @ahayden.  The new Amazon linux 2 AMI on the amazon UI. `ami-0b898040803850657`.  This AMI specified `ami-0de53d8956e8dcf80` has 14 security updates needed which you can do with `sudo yum update`

Comment (zaro0508) [reply config/prod/iwanna-instance.yaml]: updated to with the latest AMI

**Comment by zaro0508** [comment]
GH PR based provisioner has been deprecated, please use the service catalog at https://sc.sageit.org

**Pull request closed without merging**